decision_service/rlclientlib/logger/async_batcher.h

```diff
@@ -57,11 +58,9 @@ namespace reinforcement_learning {
       }
       catch(const std::exception& e) {
         _thread_is_running = false;
-        std::ostringstream os;
-        os  << "Unable to start background thread to log events. "
-            << e.what();
-        api_status::try_update(status, error_code::background_thread_start, os.str().c_str());
-        return error_code::background_thread_start;
+        return report_error(status, error_code::background_thread_start,
+                            "Unable to start background thread to log events. ",
+                            e.what());
```

Comment:
I looked at the macro and it looks fancy, but the usage is kind of unexpected. can't we do something like report_error(status, error_code::...) << "Unable to start.." << e.what()? which looks more natural?

Reply:
Yes thought about that, but report_error() would have to return a reference to object with overloaded << operator but then it also needs to return an error code.

---
In reply to: [193497746](https://github.com/JohnLangford/vowpal_wabbit/pull/1495#discussion_r193497746) [](ancestors = 193497746)